crates/aptos/src/update/prover_dependency.rs

```diff
@@ -0,0 +1,363 @@
+        };
+
+        let target = target_res?;
+
+        let install_dir = match self.install_dir.clone() {
+            Some(dir) => dir,
+            None => {
+                let dir = get_additional_binaries_dir();
+                // Make the directory if it doesn't already exist.
+                std::fs::create_dir_all(&dir)
+                    .with_context(|| format!("Failed to create directory: {:?}", dir))?;
+                dir
+            },
+        };
+
+        let dir_name = format!("{}-{}", target, arch);
+
+        BoogieOptions::check_version_is_compatible(
+            BOOGIE_BINARY_NAME,
+            TARGET_BOOGIE_VERSION,
+            MIN_BOOGIE_VERSION,
+            MAX_BOOGIE_VERSION,
+        )?;
+        self.install_binary(
+            &install_dir,
+            BOOGIE_EXE,
+            BOOGIE_BINARY_NAME,
+            TARGET_BOOGIE_VERSION,
+            "/version",
+            BOOGIE_EXE_ENV,
+            &dir_name,
+        )
+        .await?;
+
+        BoogieOptions::check_version_is_compatible(
+            Z3_BINARY_NAME,
+            TARGET_Z3_VERSION,
+            MIN_Z3_VERSION,
+            MAX_Z3_VERSION,
+        )?;
+        self.install_binary(
+            &install_dir,
+            Z3_EXE,
+            Z3_BINARY_NAME,
+            TARGET_Z3_VERSION,
+            "--version",
+            Z3_EXE_ENV,
+            &dir_name,
+        )
+        .await?;
+
+        #[cfg(unix)]
+        {
+            BoogieOptions::check_version_is_compatible(
+                CVC5_BINARY_NAME,
+                TARGET_CVC5_VERSION,
+                MIN_CVC5_VERSION,
+                MAX_CVC5_VERSION,
+            )?;
+            self.install_binary(
+                &install_dir,
+                CVC5_EXE,
+                CVC5_BINARY_NAME,
+                TARGET_CVC5_VERSION,
+                "--version",
+                CVC5_EXE_ENV,
+                &dir_name,
+            )
+            .await?;
+        }
+
+        Ok("Installation succeeded".to_string())
+    }
+
+    async fn install_binary(
+        &self,
+        path: &Path,
+        exe_name: &str,
+        binary_name: &str,
+        version: &str,
+        version_option: &str,
+        env_name: &str,
+        dir_name: &str,
+    ) -> CliTypedResult<String> {
+        let install_dir = path.join(exe_name);
+        if self
+            .check_bin_version(&install_dir, binary_name, version, version_option)
+            .is_err()
+        {
+            let download_url = format!(
+                "{}/{}/{}/raw/main/{}/{}/{}/{}",
+                GITHUB_URL, REPO_OWNER, REPO_NAME, binary_name, version, dir_name, exe_name
+            );
+            self.download_file(&download_url, &install_dir).await?;
+        }
+        if self.add_env_var(env_name, &install_dir).is_err() {
+            println!(
+                "Failed to set the environment variable {}. Please set it manually",
+                env_name
+            );
+        }
```

Comment:
1. Use `eprintln`.
2. Show the error as part of the message `{:#}`.

Reply:
done